### PR TITLE
PC-1058 Add validation to phone number entry

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1058)

# Description

adds a check to the schema for phone numbers to ensure that it is valid. uses the approved phonenumbers library to do this. uses their lower level of check; `is_possible_number()`, to reject completely impossible numbers

adds a verifying test too with some phone numbers to accept or reject

pending translation response from DESNZ

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`

# Screenshots

image

### CODE REVIEW

File: tests/test_frontdoor.py [hunk 1]
```diff
@@ -1526,6 +1526,46 @@ def test_on_check_answers_page_changing_to_no_loft_hides_loft_follow_up_question
...
 
+@pytest.mark.parametrize(
+    ("contact_number", "expect_to_accept"),
+    [
+        ("01632 960 001", True),
+        ("07700 900 982", True),
+        ("+44 808 157 0192", True),
+        ("+33 06 12 34 56 78", True),
+        ("1", False),
+        ("not a number", False),
```
Comment (jdgage) [inline tests/test_frontdoor.py:1537]: I'd really like this to have quite thorough tests. There's a bunch of valid UK numbers [here](https://gist.github.com/edwardhorsford/a9df6b16a561cd54336fbba51572db25) (linked from one of the GDS discussions) - could we cover all of those?

**Review by jdgage — CHANGES_REQUESTED** [review]
Looks good, pending translations

File: tests/test_frontdoor.py (hunk 1 above)
Comment (samyou-softwire) [reply tests/test_frontdoor.py:1537]: sure

File: tests/test_frontdoor.py [hunk 2]
```diff
@@ -1529,11 +1529,43 @@ def test_on_check_answers_page_changing_to_no_loft_hides_loft_follow_up_question
 @pytest.mark.parametrize(
     ("contact_number", "expect_to_accept"),
     [
```
Comment (jdgage) [inline tests/test_frontdoor.py:1531]: Maybe a link to that gist here would be good

**Review by jdgage — APPROVED** [review]

File: help_to_heat/locale/cy/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -454,15 +454,17 @@ msgstr "Rhowch god post dilys yn y Deyrnas Unedig"
...
 msgstr "Fformat ebost annilys"
 
-#: help_to_heat/frontdoor/schemas.py:788 help_to_heat/frontdoor/schemas.py:790
-msgid "Invalid contact number"
-msgstr "Rhif ffôn annilys"
+#: help_to_heat/frontdoor/schemas.py:789 help_to_heat/frontdoor/schemas.py:793
+msgid ""
+"Enter a telephone number, like 01632 960 001, 07700 900 982 or +44 808 157 "
+"0192"
+msgstr "Rhowch rif ffôn, fel 01632 960 001, 07700 900 982 neu +44 808 157 0192"
```
Comment (jdgage) [inline help_to_heat/locale/cy/LC_MESSAGES/django.po:461]: In the end I could have probably translated this one 😅

**Pull request merged**